url.py

```diff
@@ -0,0 +1,28 @@
+def remove_last_slash(url:str)->str:                    #Removes the last forward slash that is redundant to the url
+    if(url[-1]=='/'):
+        return(url[:-1])
+    else: 
+        return url 
+def get_protocol(url:str)->str:                         #Gets the protocol specified in the url
+    parsed_url = url.split('/')
+    protocol = parsed_url[0].replace(":",'')
+    return protocol
+
+def get_domain(url:str)->str:                           #Gets the domain in the url
+    parsed_url = url.split('/')   
+    return parsed_url[2]
+
+def get_path(url:str)->str:                             #Gets the server directory path
+    parsed_url = url.split('/',3)  
+    path = parsed_url[3].split('?')
+    return path[0]
```

Comment:
> Gets the server directory path
URL paths are not necessarily directory path on server. (it used to be so in old days but now it is not the case mostly, please check. https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_is_a_URL#path_to_resource)

Accordingly please correct the comments.

Secondly, please cover all valid URL cases. (some examples given above)